DependencyInjection/Configuration.php

```diff
@@ -371,6 +372,17 @@ private function addSharedPluginNodes(ArrayNodeDefinition $pluginNode, $disableA
                         ->children()
                             ->scalarNode('default_ttl')->defaultValue(0)->end()
                             ->scalarNode('respect_cache_headers')->defaultNull()->end()
+                            ->variableNode('respect_response_cache_directives')
+                                ->validate()
+                                ->always(function ($v) {
+                                    if (is_null($v) || is_array($v)) {
```

Comment:
if it must always be an array, use prototyped array node rather than a variable node. It will allow better handling of the data.

Rule of thumb: if you start using `variableNode`, rethink what you are doing. Most of the time, you should not need it.